fmn/rules/services/distgit.py

```diff
@@ -19,16 +19,17 @@ def __init__(self, url):
         self.url = url
         self.proxy = PagureSyncProxy(base_url=url)
 
-    @cache.cache_on_arguments()
-    def get_owners(self, artifact_type, artifact_name, user_or_group):
-        # cache this for a reasonable time
-        result = self.proxy.get(f"{artifact_type}/{artifact_name}")
-        if user_or_group == "user":
-            return result["access_users"]["owner"]
-        elif user_or_group == "group":
-            return result["access_groups"]["admin"] + result["access_groups"]["commit"]
-        else:
-            raise ValueError("Argument user_or_group must be either user or group, duh.")
+    def get_user_owners(self, artifact_type: str, artifact_name: str) -> list[str]:
+        return self.proxy.get_project_users(
+            f"{artifact_type}/{artifact_name}",
+            PagureRole.OWNER | PagureRole.ADMIN | PagureRole.COMMIT,
```

Comment:
Huh, I did not, know that OR'ed IntEnums could be iterated on to get the components!

In fact, after some research, it seems to be a new feature of Python 3.11. Can you confirm? I think we want to target 3.10, no?

Reply:
You can iterate over the type and get the constituent enum values, this is what's done in `PagureAsyncProxy.get_project_users()` and it seems to work in Python 3.10, too (same output in Python 3.11):

```
nils@makake:~> python3.10
Python 3.10.9 (main, Dec 20 2022, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import IntEnum, auto
>>> class Foo(IntEnum):
...     bar = auto()
...     baz = auto()
...     gna = auto()
... 
>>> Foo.bar | Foo.baz 
3
>>> for x in Foo:
...     print(x)
... 
Foo.bar
Foo.baz
Foo.gna
>>>
```